### PR TITLE
Small updates to enable Cython 3

### PR DESCRIPTION
**Pull request by hmaarrfk**
Mostly it seems that cython wants you to declare things as `noexcept`

Closes https://github.com/PyAV-Org/PyAV/issues/1140

I'm going to see if I can report the bug to cython, but it is hard for me to create a minimum reproducible example

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Tested locally and cython 0.29.36 and it seems to compile and function just as well.

File: av/video/format.pyx [hunk 1]
```diff
@@ -47,10 +47,14 @@ cdef class VideoFormat(object):
         self.ptr = lib.av_pix_fmt_desc_get(pix_fmt)
         self.width = width
         self.height = height
-        self.components = tuple(
+        # hmaarrfk -- 2023/07/23
+        # Note on tuple([])
+        # Cython 3 seems to have trouble with cdef tuples, so we use a list
+        # it complains about some const identifier not being able to get assigned
```
Comment (SpecLad) [inline av/video/format.pyx:53]: FWIW, this issue was fixed in Cython 3.0.1, so this workaround can probably be removed.

**Comment by moi90** [comment]
It works for me locally, too. I hope this gets merged soon!

**Comment by rvanlaar** [comment]
What's needed to get this merged?

**Comment by jlaine** [comment]
> What's needed to get this merged?

Co-maintainers!

File: av/video/format.pyx (hunk 1 above)
Comment (jlaine) [reply av/video/format.pyx:53]: If this isn't needed can we prune it please?

File: docs/development/includes.py [hunk 2]
```diff
@@ -107,9 +107,16 @@ def extract(path, **kwargs):
         c_string_encoding='ascii',
     )
 
-    context = options.create_context()
+    context = Context(
```
Comment (jlaine) [inline docs/development/includes.py:110]: How do these changes relate to the PR?

**Comment by jlaine** [comment]
Can you point us to some Cython docs explaining why these changes are required, and if there are others we need to look out for?

File: docs/development/includes.py (hunk 2 above)
Comment (hmaarrfk) [reply docs/development/includes.py:110]: cython3 removed create_context, not sure when so I looked at the old implementation of create_context and copied it here.

**Comment by hmaarrfk** [comment]
I'm not sure there is much to point to:

1. Cython wanted to optimize more so they ask that you write `noexcept`. Search for `noexcept` in https://cython.readthedocs.io/en/latest/src/userguide/language_basics.html
2. There was a bug that I reported upstream https://github.com/cython/cython/issues/5558 that got fixed.
3. I'm not sure how I found the `create_context` was removed. sometimes I dig deep into code. I don't have a good reference for how I did this months ago

**Comment by SpecLad** [comment]
> Can you point us to some Cython docs explaining why these changes are required, and if there are others we need to look out for?

It's explained here: https://cython.readthedocs.io/en/latest/src/userguide/migrating_to_cy30.html#exception-values-and-noexcept

**Comment by hmaarrfk** [comment]
i'm not sure why the tests are failing.

File: docs/development/includes.py (hunk 2 above)
Comment (ENate) [reply docs/development/includes.py:110]: Do we have to add this in the dev folder? There no such folder available inside transformers

**Pull request merged**